Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -109,73 +128,111 @@ public static string FindMostPoorPerson(string[] peopleAndBalances)
                 }
-                else if (person.Balance[^1] == lowerBalance)
+            }
+
+            int mostPoorArraySize = 0;
+            for (int i = 0; i < mostPoorPersonArray.Length; i++)
+            {
+                if (mostPoorPersonArray[i + 1] == "" || mostPoorPersonArray[i + 1] == null)
                 {
-                    mostPoorPerson.Add(person.Name);
+                    mostPoorArraySize = i + 1;
+                    break;
                 }
             }
-            return $"{FormatName(mostPoorPerson)} {(mostPoorPerson.Count > 1 ? "have the least money." : "has the least money.")} {(lowerBalance >= 0 ? null : "-")}¤{(Math.Abs(lowerBalance).ToString())}.";
+
+            return $"{FormatNameArray(mostPoorPersonArray)} {(mostPoorArraySize > 1 ? "have the least money." : "has the least money.")} {(lowerBalance >= 0 ? null : "-")}¤{(Math.Abs(lowerBalance).ToString())}.";
         }
 
-        public static List<Person> GetPeople(string[] peopleAndBalances)
+        public static Person[] GetPeopleArray(string[] peopleAndBalances)
         {
-            List<Person> People = new List<Person>();
-            foreach (string peopleAndBalance in peopleAndBalances)
+            Person[] People = new Person[peopleAndBalances.Length];
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
             {
-                var irregularArray = peopleAndBalance.Split(',');
+                var irregularArray = peopleAndBalances[i].Split(',');
                 Person person = new Person();
                 person.Name = irregularArray[0];
-                for (int i = 1; i < irregularArray.Length; i++)
+
+                for (int j = 1; j < irregularArray.Length; j++)
                 {
                     decimal balance;
-                    if (decimal.TryParse(irregularArray[i], NumberStyles.Any, new CultureInfo("en-US"), out balance))
+                    if (decimal.TryParse(irregularArray[j], NumberStyles.Any, new CultureInfo("en-US"), out balance))
                     {
-                        person.Balance.Add(balance);
+                        person.Balance[j - 1] = balance;
                     }
                 }
-                People.Add(person);
+
+                People[i] = person;
             }
+
             return People;
         }
-
-        private static string FormatName(List<string> Name)
+        
+        private static string FormatNameArray(string[] nameArray)
         {
+            if (nameArray[1] == "" || nameArray[1] == null)
+                return nameArray[0];
             string formattedName = "";
-            if (Name.Count == 1)
-                return Name[0];
 
-            for (int i = 0; i < Name.Count; i++)
+            for (int i = 0; i < nameArray.Length; i++)
             {
-                formattedName += Name[i];
-                if (Name.Count - 2 == i)
+                formattedName += nameArray[i];
```

Comment:
Instead of concatinating a string like this, use `StringBuilder`. The performance difference, starting from 5 iterations starts changing in multiple times!